File: models/previewCardSqlModel.ts
import {
  DataTypes,
  Model,
  ModelStatic
} from 'sequelize'
import sequelize from '@/config/sequelize'
import PreviewCardSqlRecord from '@/types/PreviewCardSqlRecord'
import createId from '@/utilities/createId'
const previewCardSqlModel: ModelStatic<Model<PreviewCardSqlRecord>> = sequelize.models.PreviewCard ?? sequelize.define<Model<PreviewCardSqlRecord>>(
  'PreviewCard', {
    ...createId(),
    url: {
      type: DataTypes.TEXT,
      allowNull: false,
      unique: true,
      validate: {
        isUrl: true
      }
    },
    title: {
      type: DataTypes.TEXT,
      allowNull: false
    },
    description: {
      type: DataTypes.TEXT,
      allowNull: false
    },
    type: {
      type: DataTypes.BIGINT,
      allowNull: false
    },
    html: {
      type: DataTypes.TEXT,
      allowNull: false
    },
    author_name: {
      type: DataTypes.TEXT,
      allowNull: false
    },
    author_url: {
      type: DataTypes.TEXT,
      allowNull: false
    },
    provider_name: {
      type: DataTypes.TEXT,
      allowNull: false
    },
    provider_url: {
      type: DataTypes.TEXT,
      allowNull: false,
      validate: {
        isUrl: true
      }
    },
    width: {
      type: DataTypes.FLOAT,
      allowNull: false
    },
    height: {
      type: DataTypes.FLOAT,
      allowNull: true
    },
    embed_url: {
      type: DataTypes.TEXT,
      allowNull: false,
      validate: {
        isUrl: true
      }
    },
    image_file_name: {
      type: DataTypes.TEXT,
      allowNull: true
    },
    image_content_type: {
      type: DataTypes.TEXT,
      allowNull: true
    },
    image_file_size: {
      type: DataTypes.FLOAT,
      allowNull: true
    },
    image_updated_at: {
      type: DataTypes.DATE,
      allowNull: true
    }
  }, {
    tableName: 'preview_cards',
    timestamps: true
  }
)
export default previewCardSqlModel

File: nomad/models/preview_card_sql_model.py
"""
Preview card SQL table model
"""
from uuid import uuid4
from django.db import models
class PreviewCard(models.Model):
    """
    Preview card SQL table model
    """
    class Meta:
        """
        Preview card SQL table model meta class
        """
        verbose_name = 'Preview Card'
        verbose_name_plural = f'{verbose_name}s'
        db_table = verbose_name_plural.lower().replace(
            ' ',
            '_'
        )
    id = models.UUIDField(
        primary_key = True,
        default = uuid4,
        editable = False
    )
    url = models.URLField(
        unique = True
    )
    title = models.TextField()
    description = models.TextField()
    type = models.PositiveSmallIntegerField()
    html = models.TextField()
    author_name = models.TextField()
    author_url = models.URLField()
    provider_name = models.TextField()
    provider_url = models.URLField()
    width = models.PositiveBigIntegerField()
    height = models.PositiveSmallIntegerField(
        null = True,
        blank = True
    )
    embed_url = models.URLField()
    image_file_name = models.TextField(
        null = True,
        blank = True
    )
    image_content_type = models.TextField(
        null = True,
        blank = True
    )
    image_file_size = models.PositiveSmallIntegerField(
        null = True,
        blank = True
    )
    image_updated_at = models.DateTimeField(
        null = True,
        blank = True
    )
    created_at = models.DateTimeField(
        auto_now_add = True
    )
    updated_at = models.DateTimeField(
        auto_now = True
    )
